### PR TITLE
add e2e test for teal script

### PR DESCRIPTION
**Pull request by algobolson**
exercises several cases of TLHC pattern with goal commands

### CODE REVIEW

File: cmd/goal/clerk.go [hunk 1]
```diff
@@ -566,6 +571,17 @@ func getProto(versArg string) config.ConsensusParams {
 	cvers := protocol.ConsensusCurrentVersion
 	if versArg != "" {
 		cvers = protocol.ConsensusVersion(versArg)
+	} else {
+		dataDir := maybeSingleDataDir()
+		if dataDir != "" {
+			client := ensureAlgodClient(dataDir)
+			params, err := client.SuggestedParams()
```
Comment (tsachiherman) [inline cmd/goal/clerk.go]: I think that you don't want this one for a dryrun

Comment (algobolson) [reply cmd/goal/clerk.go]: I think it doesn't hurt to have it. It will try to contact an algod but fall back to defaults if an algod can't be found.

Comment (tsachiherman) [reply cmd/goal/clerk.go]: My concern is not that we would try and fail. My concern is that connecting to a live node is opposed to "dryrun" mode. Do you think we could skip this portion for when we specify dryrun ?

**Comment by tsachiherman** [comment]
Could you add some more printing so that in case of an error, we will be able to associate this particular test with the error string ? ( or the 'OK' result ? )

File: cmd/goal/clerk.go (hunk 1 above)
Comment (algobolson) [reply cmd/goal/clerk.go]: If you have $ALGORAND_DATA defined and this quietly checks with your network for the current version, that sounds like a good thing to me. `--proto` preempts and overrides that so the check with algod is never done. I think this is a good way to guess a reasonable default.

**Comment by algobolson** [comment]
the bash script starts with `set -x` so that it echos every command it runs. we'll know which one failed.

**Comment by algobolson** [comment]
ping. I think this is stable and good to go.

**Pull request merged**